x-pack/plugins/maps/public/connected_components/mb_map/mb_map.tsx

```diff
@@ -131,6 +133,35 @@ export class MBMap extends Component<Props, State> {
     this.props.onMapDestroyed();
   }
 
+  // This keeps track of the latest update calls, per layerId
+  _queryForMeta = (() => {
+    const callMap = new Map<string, ILayer>();
+
+    let timerId: ReturnType<typeof setTimeout> | null = null;
+
+    const updateAllLayers = () => {
+      callMap.forEach((layer: ILayer, layerId: string) => {
+        if (this.state.mbMap) {
+          if (layer.isVisible()) {
+            const mbFeatures = layer.queryForTileMeta(this.state.mbMap);
+            if (mbFeatures !== null) {
+              this.props.updateCounts(layerId, mbFeatures);
+            }
+          }
+        }
+      });
+      callMap.clear();
+      timerId = null;
+    };
+
+    return (layerId: string, layer: ILayer) => {
+      callMap.set(layerId, layer);
+      if (!timerId) {
+        timerId = setTimeout(updateAllLayers, 1200);
```

Comment:
Where did 1200 come from? Can you explain this a bit?

Reply:
No real reason. Some trial and error. Trying to find a balance between long-enough so we don't re-evaluate tile-meta too often, but not too long so that the user does not get appropriate visual feedback.

Reply:
Why is there a timer at all? Shouldn't updateCounts just be called whenever a new tile is loaded?